<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.340">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>CDRL Kinome Analysis Omnibus - 4&nbsp; Analytical Approaches</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./upstream_kinase.html" rel="next">
<link href="./data_generation.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./analytical_approaches.html"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Analytical Approaches</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">CDRL Kinome Analysis Omnibus</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./platform.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">The PamStation® Platform</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data_generation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data Generation</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./analytical_approaches.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Analytical Approaches</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./upstream_kinase.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Upstream Kinase Identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_pathways.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Pathway Analysis of Kinase Data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_visualization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Visualizing Kinase Data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./case_study.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Case Study</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#approaches" id="toc-approaches" class="nav-link active" data-scroll-target="#approaches"><span class="header-section-number">4.1</span> Approaches</a></li>
  <li><a href="#upstream-kinase-identification" id="toc-upstream-kinase-identification" class="nav-link" data-scroll-target="#upstream-kinase-identification"><span class="header-section-number">4.2</span> Upstream Kinase Identification</a></li>
  <li><a href="#pathway-analysis" id="toc-pathway-analysis" class="nav-link" data-scroll-target="#pathway-analysis"><span class="header-section-number">4.3</span> Pathway Analysis</a></li>
  <li><a href="#visualization-and-summarization" id="toc-visualization-and-summarization" class="nav-link" data-scroll-target="#visualization-and-summarization"><span class="header-section-number">4.4</span> Visualization and Summarization</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Analytical Approaches</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>The PamStation® 12 Platform generates a large amount of data in a single experiment. The raw data obtained from the platform includes fluorescence intensity values for each well, at each cycle and exposure level. This data is rich in information, but in its raw form, it cannot be easily interpreted without further processing. The preprocessing step transforms the raw image data into fluorescence intensity values that can be analyzed using various computational methods. In this chapter, we outline different analytical approaches that can be taken to extract meaningful insights from the processed data. These approaches include upstream kinase analysis, unsupervised and supervised clustering methods, statistical approaches, pathway analysis, and visualization techniques. We also discuss the advantages and limitations of each approach. Through the use of these analytical approaches, we can uncover hidden patterns and relationships in the data and gain a deeper understanding of kinase signaling pathways.</p>
<p><a href="case_study.html">Chapter&nbsp;<span>8</span></a> will walk us through all of the approaches outlined here on an example dataset.</p>
<section id="approaches" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="approaches"><span class="header-section-number">4.1</span> Approaches</h2>
<p>The analysis of data obtained from the PamStation® 12 platform can help answer two key questions: which kinases are the most important in a given sample, and what functional consequences arise from perturbing a particular kinase? Although a well-designed experiment can address both questions simultaneously, it is also possible to conduct separate experiments for each. To analyze the data and derive meaningful insights, a range of analytical techniques can be employed.</p>
<p>These techniques can be broadly divided into three categories:</p>
<ol type="1">
<li>Upstream kinase identification techniques</li>
<li>Kinase pathway analysis techniques</li>
<li>Visualization and summarization techniques.</li>
</ol>
<p>Upstream kinase identification techniques seek to identify the kinases that are responsible for the observed changes in phosphorylation levels, typically by perturbing the system and then using statistical methods to identify the kinases whose activity changes significantly.</p>
<p>Kinase pathway analysis techniques aim to identify the pathways that are affected by kinase perturbations, using either experimental data or existing knowledge of kinase signaling networks.</p>
<p>Visualization and summarization techniques enable the visualization of complex data sets and summarization of key findings, providing a means of communicating results to others.</p>
<p>Each of these categories includes several techniques that can be tailored to the specific research question and experimental design. The following sections will elaborate on each category and provide examples of the techniques that fall within each.</p>
</section>
<section id="upstream-kinase-identification" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="upstream-kinase-identification"><span class="header-section-number">4.2</span> Upstream Kinase Identification</h2>
<p>The raw output from the preprocessing step is phosphorylation status of individual peptides. This only indirectly measures the activity of the kinases by way of the identificaiton of targets of individual kinases. Thus, it is important to have a way to identify indivdual kinases that are upstream of a particula peptide and thus, may be involved in the phosphorylation of such targets.</p>
<p>The general approach to upstream kinase identification can be summarised as follows:</p>
<ol type="1">
<li>Select a particular state of phosphorylation from the input data. Usually this is the phosphorylation data from the final cycle of the array run.</li>
<li>For each individual sample, define a variable that will summarise the phosphorylation level of that peptide.</li>
<li>Identify how the phosphorylation status changes with respect to a control group</li>
<li>Use either experimental, literature-based or computational predictions for possible kinases upstream of a particular peptide and use a statistical model to identify the most prominent ones.</li>
</ol>
<p>There are four available packages that allow us to do upstream kinase identification. These are:</p>
<ul>
<li>Upstream Kinase Analysis (UKA)</li>
<li>Kinome Random Sampling Analysis (KRSA)</li>
<li>Kinase Enrichment Analysis v3 (KEA3)</li>
<li>Post-Translational Modification Signature Enrichment Analysis (PTM-SEA)</li>
</ul>
</section>
<section id="pathway-analysis" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="pathway-analysis"><span class="header-section-number">4.3</span> Pathway Analysis</h2>
<p>While identification of kinases is important and useful in its own right, understanding the functional consequences of the kinase activity is equally, if not more important. For this purpose, we can utilize various databases that have collected and curated functional information about the genes. This question makes the core of pathway analysis <span class="citation" data-cites="khatri2012">(<a href="references.html#ref-khatri2012" role="doc-biblioref">Khatri, Sirota, and Butte 2012</a>)</span>.</p>
<p>For pathway analysis, there are two different approaches that rely on different results: utilizing the substrate activity for identifying downstream pathways or utilizing the identified kinases for identifying downstream pathways.</p>
<p>For both cases, the general workflow is as follows:</p>
<ol type="1">
<li>Identify the genes that have been impacted in the experiment, either by a threshold or by a proportional measure.</li>
<li>Identify the relevant database that has the information you need.</li>
<li>Use the particular database’s interface to upload the gene data and scores if necessary to get the list of enriched pathways.</li>
</ol>
<p>Several pathway analysis tools are available, such as Reactome <span class="citation" data-cites="gillespie2021 griss2020 jassal2019 fabregat2018 fabregat2017 sidiropoulos2017 fabregat2017a wu2017">(<a href="references.html#ref-gillespie2021" role="doc-biblioref">Gillespie et al. 2021</a>; <a href="references.html#ref-griss2020" role="doc-biblioref">Griss et al. 2020</a>; <a href="references.html#ref-jassal2019" role="doc-biblioref">Jassal et al. 2019</a>; <a href="references.html#ref-fabregat2018" role="doc-biblioref">Fabregat et al. 2018</a>; <a href="references.html#ref-fabregat2017" role="doc-biblioref">Fabregat, Sidiropoulos, Viteri, Marin-Garcia, et al. 2017</a>; <a href="references.html#ref-sidiropoulos2017" role="doc-biblioref">Sidiropoulos et al. 2017</a>; <a href="references.html#ref-fabregat2017a" role="doc-biblioref">Fabregat, Sidiropoulos, Viteri, Forner, et al. 2017</a>; <a href="references.html#ref-wu2017" role="doc-biblioref">Wu and Haw 2017</a>)</span>, KEGG <span class="citation" data-cites="kanehisa2000 kanehisa2019 kanehisa2022">(<a href="references.html#ref-kanehisa2000" role="doc-biblioref">M. Kanehisa 2000</a>; <a href="references.html#ref-kanehisa2019" role="doc-biblioref">Minoru Kanehisa 2019</a>; <a href="references.html#ref-kanehisa2022" role="doc-biblioref">Minoru Kanehisa et al. 2022</a>)</span>, and GO enrichment analysis <span class="citation" data-cites="ashburner2000 thegene2020">(<a href="references.html#ref-ashburner2000" role="doc-biblioref">Ashburner et al. 2000</a>; <a href="references.html#ref-thegene2020" role="doc-biblioref">Carbon et al. 2020</a>)</span>. These databases provide curated pathway annotations and functional analysis of the genes, making it easier to interpret the results obtained from PamGene data analysis.</p>
<p>Reactome is a database of manually curated pathway information, and it provides a web-based interface for the analysis of gene lists. The software can analyze the gene list and identify enriched pathways based on the input data <span class="citation" data-cites="gillespie2021 griss2020 jassal2019 fabregat2018 fabregat2017 sidiropoulos2017 fabregat2017a wu2017">(<a href="references.html#ref-gillespie2021" role="doc-biblioref">Gillespie et al. 2021</a>; <a href="references.html#ref-griss2020" role="doc-biblioref">Griss et al. 2020</a>; <a href="references.html#ref-jassal2019" role="doc-biblioref">Jassal et al. 2019</a>; <a href="references.html#ref-fabregat2018" role="doc-biblioref">Fabregat et al. 2018</a>; <a href="references.html#ref-fabregat2017" role="doc-biblioref">Fabregat, Sidiropoulos, Viteri, Marin-Garcia, et al. 2017</a>; <a href="references.html#ref-sidiropoulos2017" role="doc-biblioref">Sidiropoulos et al. 2017</a>; <a href="references.html#ref-fabregat2017a" role="doc-biblioref">Fabregat, Sidiropoulos, Viteri, Forner, et al. 2017</a>; <a href="references.html#ref-wu2017" role="doc-biblioref">Wu and Haw 2017</a>)</span>.</p>
<p>KEGG is another database for understanding high-level functions and utilities of the biological system, such as the cell, the organism, and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies <span class="citation" data-cites="kanehisa2000 kanehisa2019 kanehisa2022">(<a href="references.html#ref-kanehisa2000" role="doc-biblioref">M. Kanehisa 2000</a>; <a href="references.html#ref-kanehisa2019" role="doc-biblioref">Minoru Kanehisa 2019</a>; <a href="references.html#ref-kanehisa2022" role="doc-biblioref">Minoru Kanehisa et al. 2022</a>)</span>.</p>
<p>GO enrichment analysis is another pathway analysis technique that is widely used. It is used to determine whether certain GO terms (i.e., biological processes, cellular components, and molecular functions) are overrepresented in the list of genes. This method helps to identify which GO terms are most impacted by the gene list and can provide a comprehensive understanding of the biological context of the analyzed data <span class="citation" data-cites="ashburner2000 thegene2020">(<a href="references.html#ref-ashburner2000" role="doc-biblioref">Ashburner et al. 2000</a>; <a href="references.html#ref-thegene2020" role="doc-biblioref">Carbon et al. 2020</a>)</span>.</p>
</section>
<section id="visualization-and-summarization" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="visualization-and-summarization"><span class="header-section-number">4.4</span> Visualization and Summarization</h2>
<p>In analyzing the large amount of data generated by the primary data and subsequent analyses, it is often necessary to summarize the data in a figure to aid in interpretation. One commonly used way to summarize global kinase activity is through the use of a row-scaled heatmap. This type of heatmap allows for the clustering of both samples and peptides, which can identify homogeneity within samples and functional clusters of peptides. This is particularly useful for analyzing activity across all the samples and peptides present on the chip.</p>
<p>Another effective way to summarize global activity is through boxplots or violin plots of all the peptides in a given sample. These plots can be split by barcodes and group membership to showcase differences and heterogeneity, respectively. This approach allows for the visualization of the distribution of activity across peptides and provides insight into the level of variation within each sample.</p>
<p>For group comparisons, a waterfall plot can be used to show the change in activity between two groups. This type of plot shows individual changes in peptide activity between two groups, with the change on the x-axis and the peptides on the y-axis. Replicates, if present, can be plotted on the same line with an additional “mean” change dot plotted in the middle. This type of plot can help to identify differences in activity between groups and highlight potential biomarkers.</p>
<p>Heatmaps are also useful for group comparisons, where only the samples of interest are kept and the data is normalized among them. These group-specific heatmaps are similar to the global heatmaps but contain only the two designated groups. This allows for an easier comparison between the two groups, while also highlighting differences in activity. Overall, the choice of figure type will depend on the research question and the type of data being analyzed.</p>


<div id="refs" class="references csl-bib-body hanging-indent" role="list" style="display: none">
<div id="ref-ashburner2000" class="csl-entry" role="listitem">
Ashburner, Michael, Catherine A. Ball, Judith A. Blake, David Botstein, Heather Butler, J. Michael Cherry, Allan P. Davis, et al. 2000. <span>“Gene Ontology: Tool for the Unification of Biology.”</span> <em>Nature Genetics</em> 25 (1): 25–29. <a href="https://doi.org/10.1038/75556">https://doi.org/10.1038/75556</a>.
</div>
<div id="ref-thegene2020" class="csl-entry" role="listitem">
Carbon, Seth, Eric Douglass, Benjamin M Good, Deepak R Unni, Nomi L Harris, Christopher J Mungall, Siddartha Basu, et al. 2020. <span>“The Gene Ontology Resource: Enriching a GOld Mine.”</span> <em>Nucleic Acids Research</em> 49 (D1): D325–34. <a href="https://doi.org/10.1093/nar/gkaa1113">https://doi.org/10.1093/nar/gkaa1113</a>.
</div>
<div id="ref-fabregat2018" class="csl-entry" role="listitem">
Fabregat, Antonio, Florian Korninger, Guilherme Viteri, Konstantinos Sidiropoulos, Pablo Marin-Garcia, Peipei Ping, Guanming Wu, Lincoln Stein, Peter D’Eustachio, and Henning Hermjakob. 2018. <span>“Reactome Graph Database: Efficient Access to Complex Pathway Data.”</span> Edited by Timothée Poisot. <em>PLOS Computational Biology</em> 14 (1): e1005968. <a href="https://doi.org/10.1371/journal.pcbi.1005968">https://doi.org/10.1371/journal.pcbi.1005968</a>.
</div>
<div id="ref-fabregat2017a" class="csl-entry" role="listitem">
Fabregat, Antonio, Konstantinos Sidiropoulos, Guilherme Viteri, Oscar Forner, Pablo Marin-Garcia, Vicente Arnau, Peter D’Eustachio, Lincoln Stein, and Henning Hermjakob. 2017. <span>“Reactome Pathway Analysis: A High-Performance in-Memory Approach.”</span> <em>BMC Bioinformatics</em> 18 (1). <a href="https://doi.org/10.1186/s12859-017-1559-2">https://doi.org/10.1186/s12859-017-1559-2</a>.
</div>
<div id="ref-fabregat2017" class="csl-entry" role="listitem">
Fabregat, Antonio, Konstantinos Sidiropoulos, Guilherme Viteri, Pablo Marin-Garcia, Peipei Ping, Lincoln Stein, Peter D’Eustachio, and Henning Hermjakob. 2017. <span>“Reactome Diagram Viewer: Data Structures and Strategies to Boost Performance.”</span> Edited by Janet Kelso. <em>Bioinformatics</em> 34 (7): 1208–14. <a href="https://doi.org/10.1093/bioinformatics/btx752">https://doi.org/10.1093/bioinformatics/btx752</a>.
</div>
<div id="ref-gillespie2021" class="csl-entry" role="listitem">
Gillespie, Marc, Bijay Jassal, Ralf Stephan, Marija Milacic, Karen Rothfels, Andrea Senff-Ribeiro, Johannes Griss, et al. 2021. <span>“The Reactome Pathway Knowledgebase 2022.”</span> <em>Nucleic Acids Research</em> 50 (D1): D687–92. <a href="https://doi.org/10.1093/nar/gkab1028">https://doi.org/10.1093/nar/gkab1028</a>.
</div>
<div id="ref-griss2020" class="csl-entry" role="listitem">
Griss, Johannes, Guilherme Viteri, Konstantinos Sidiropoulos, Vy Nguyen, Antonio Fabregat, and Henning Hermjakob. 2020. <span>“ReactomeGSA - Efficient Multi-Omics Comparative Pathway Analysis.”</span> <em>Molecular &amp; Cellular Proteomics</em> 19 (12): 2115–25. <a href="https://doi.org/10.1074/mcp.tir120.002155">https://doi.org/10.1074/mcp.tir120.002155</a>.
</div>
<div id="ref-jassal2019" class="csl-entry" role="listitem">
Jassal, Bijay, Lisa Matthews, Guilherme Viteri, Chuqiao Gong, Pascual Lorente, Antonio Fabregat, Konstantinos Sidiropoulos, et al. 2019. <span>“The Reactome Pathway Knowledgebase.”</span> <em>Nucleic Acids Research</em>, November. <a href="https://doi.org/10.1093/nar/gkz1031">https://doi.org/10.1093/nar/gkz1031</a>.
</div>
<div id="ref-kanehisa2000" class="csl-entry" role="listitem">
Kanehisa, M. 2000. <span>“KEGG: Kyoto Encyclopedia of Genes and Genomes.”</span> <em>Nucleic Acids Research</em> 28 (1): 27–30. <a href="https://doi.org/10.1093/nar/28.1.27">https://doi.org/10.1093/nar/28.1.27</a>.
</div>
<div id="ref-kanehisa2019" class="csl-entry" role="listitem">
Kanehisa, Minoru. 2019. <span>“Toward Understanding the Origin and Evolution of Cellular Organisms.”</span> <em>Protein Science</em> 28 (11): 1947–51. <a href="https://doi.org/10.1002/pro.3715">https://doi.org/10.1002/pro.3715</a>.
</div>
<div id="ref-kanehisa2022" class="csl-entry" role="listitem">
Kanehisa, Minoru, Miho Furumichi, Yoko Sato, Masayuki Kawashima, and Mari Ishiguro-Watanabe. 2022. <span>“KEGG for Taxonomy-Based Analysis of Pathways and Genomes.”</span> <em>Nucleic Acids Research</em> 51 (D1): D587–92. <a href="https://doi.org/10.1093/nar/gkac963">https://doi.org/10.1093/nar/gkac963</a>.
</div>
<div id="ref-khatri2012" class="csl-entry" role="listitem">
Khatri, Purvesh, Marina Sirota, and Atul J. Butte. 2012. <span>“Ten Years of Pathway Analysis: Current Approaches and Outstanding Challenges.”</span> Edited by Christos A. Ouzounis. <em>PLoS Computational Biology</em> 8 (2): e1002375. <a href="https://doi.org/10.1371/journal.pcbi.1002375">https://doi.org/10.1371/journal.pcbi.1002375</a>.
</div>
<div id="ref-sidiropoulos2017" class="csl-entry" role="listitem">
Sidiropoulos, Konstantinos, Guilherme Viteri, Cristoffer Sevilla, Steve Jupe, Marissa Webber, Marija Orlic-Milacic, Bijay Jassal, et al. 2017. <span>“Reactome Enhanced Pathway Visualization.”</span> Edited by Janet Kelso. <em>Bioinformatics</em> 33 (21): 3461–67. <a href="https://doi.org/10.1093/bioinformatics/btx441">https://doi.org/10.1093/bioinformatics/btx441</a>.
</div>
<div id="ref-wu2017" class="csl-entry" role="listitem">
Wu, Guanming, and Robin Haw. 2017. <span>“Functional Interaction Network Construction and Analysis for Disease Discovery.”</span> In, 235–53. Springer New York. <a href="https://doi.org/10.1007/978-1-4939-6783-4_11">https://doi.org/10.1007/978-1-4939-6783-4_11</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Handle positioning of the toggle
      window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
      const annoteTargets = window.document.querySelectorAll('.code-annotation-anchor');
      for (let i=0; i<annoteTargets.length; i++) {
        const annoteTarget = annoteTargets[i];
        const targetCell = annoteTarget.getAttribute("data-target-cell");
        const targetAnnotation = annoteTarget.getAttribute("data-target-annotation");
        const contentFn = () => {
          const content = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          if (content) {
            const tipContent = content.cloneNode(true);
            tipContent.classList.add("code-annotation-tip-content");
            return tipContent.outerHTML;
          }
        }
        const config = {
          allowHTML: true,
          content: contentFn,
          onShow: (instance) => {
            selectCodeLines(instance.reference);
            instance.reference.classList.add('code-annotation-active');
            window.tippy.hideAll();
          },
          onHide: (instance) => {
            unselectCodeLines();
            instance.reference.classList.remove('code-annotation-active');
          },
          maxWidth: 300,
          delay: [50, 0],
          duration: [200, 0],
          offset: [5, 10],
          arrow: true,
          trigger: 'click',
          appendTo: function(el) {
            return el.parentElement.parentElement.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto',
          placement: 'right',
          positionFixed: true,
          popperOptions: {
            modifiers: [
            {
              name: 'flip',
              options: {
                flipVariations: false, // true by default
                allowedAutoPlacements: ['right'],
                fallbackPlacements: ['right', 'top', 'top-start', 'top-end'],
              },
            },
            {
              name: 'preventOverflow',
              options: {
                mainAxis: false,
                altAxis: false
              }
            }
            ]        
          }      
        };
        window.tippy(annoteTarget, config); 
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./data_generation.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data Generation</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./upstream_kinase.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Upstream Kinase Identification</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>